charts/agent/templates/_helpers.tpl

```diff
@@ -242,6 +242,52 @@ Determine collector endpoint based on provided region
     {{- end -}}
 {{- end -}}
 
+{{/*
+Determine sysdig monitor endpoint based on provided region
+*/}}
+{{- define "monitorUrl" -}}
+    {{- if (or .Values.collectorSettings.collectorHost (eq .Values.global.sysdig.region "custom")) -}}
+        {{- required "collectorSettings.collectorHost is required for custom regions" (.Values.collectorSettings.collectorHost) -}}
+    {{- else if (eq .Values.global.sysdig.region "us1") -}}
+        {{- "app.sysdigcloud.com" -}}
+    {{- else if (eq .Values.global.sysdig.region "us2") -}}
+        {{- "us2.app.sysdig.com" -}}
+    {{- else if (eq .Values.global.sysdig.region "us3") -}}
+        {{- "us3.app.sysdig.com" -}}
+    {{- else if (eq .Values.global.sysdig.region "us4") -}}
+        {{- "app.us4.sysdig.com" -}}
+    {{- else if (eq .Values.global.sysdig.region "eu1") -}}
+        {{- "eu1.app.sysdig.com" -}}
+    {{- else if (eq .Values.global.sysdig.region "au1") -}}
+        {{- "app.au1.sysdig.com" -}}
+    {{- else -}}
+        {{- fail (printf "global.sysdig.region=%s provided is not recognized." .Values.global.sysdig.region ) -}}
+    {{- end -}}
+{{- end -}}
+
+{{/*
+Determine sysdig secure endpoint based on provided region
+*/}}
+{{- define "secureUrl" -}}
+    {{- if (or .Values.collectorSettings.collectorHost (eq .Values.global.sysdig.region "custom")) -}}
+        {{- required "collectorSettings.collectorHost is required for custom regions" (.Values.collectorSettings.collectorHost) -}}
```

Comment:
Secure also has a few places where we need to specify `apiEndpoint` as well, like https://github.com/sysdiglabs/charts/blob/master/charts/node-analyzer/values.yaml#L86. Should we make those a requirement here too?

Reply:
I see it as it should be done if <that condition> then make it required. It's no use pushing a requirement when it is not needed. For this particular case in notes it makes sense to add that requirement only if it will be used to generate the secureUrl that will be presented in the notes section. if the apiEndpoint needs to be at some other place it should be at that particular place.